main.py

```diff
@@ -99,41 +94,68 @@ async def run(
     await page.goto(url)
 
     # Focus on and click the map element
-    await page.focus('.mapboxgl-canvas')
-    await page.click('.mapboxgl-canvas')
+    await asyncio.gather(page.focus('.mapboxgl-canvas'), page.click('.mapboxgl-canvas'))
+
     await asyncio.gather(
         page.evaluate(
             """
-            () => (window.performance.mark("benchmark:start"))
+            () => (window.performance.mark("benchmark-initial-load:start"))
             """
         ),
         page.click('//div[text()="Display"]/following-sibling::button'),
     )
 
-    # Wait for the map to be idle and then stop timer
+    # Wait for the timeout to be reached
     await mark_and_measure(
-        page=page, start_mark='benchmark:start', end_mark='benchmark:end', label='bench'
+        page=page,
+        start_mark='benchmark-initial-load:start',
+        end_mark='benchmark-initial-load:end',
+        label='benchmark-initial-load',
     )
-    # Save screenshot to temporary file
-    path = screenshot_dir / f'{now}-{run_number}.png'
-    await page.screenshot(path=path)
-    print(f"[bold cyan]📸 Screenshot saved as '{path}'[/bold cyan]")
 
+    await asyncio.gather(page.focus('.mapboxgl-canvas'), page.click('.mapboxgl-canvas'))
+
+    if zoom_level:
+        for level in range(zoom_level):
+            start_mark = f'benchmark-{action}-level-{level}:start'
+            end_mark = f'benchmark-{action}-level-{level}:end'
+            label = f'benchmark-{action}-level-{level}'
+            if action == 'zoom_in':
+                await page.keyboard.press('=')
+                await page.evaluate(
+                    f"""
+                        () => (window.performance.mark("{start_mark}"))
+                        """
+                ),
```

Comment:
@katamartin, do we want these to be fired asynchronously?

Reply:
This seems `await`ing this it could create issues similar to what we saw with the page load events https://github.com/carbonplan/benchmark-maps/pull/22#issuecomment-1652619471. In this case, we would risk `performance.mark`ing after zoom-related requests/rendering cycles had already started.

I would suggest doing something similar to wait we did for hitting the display button:

```
await asyncio.gather(
    page.evaluate(
        f"""
            () => (window.performance.mark("{start_mark}"))
        """
    ),
    page.keyboard.press('=')
)